plugins/processors/ec2tagger/ec2tagger.go

```diff
@@ -418,44 +366,47 @@ func (t *Tagger) refreshLoopToUpdateTagsAndVolumes() {
 	}
 }
 
-// updateVolumes calls EC2 describe volume
-func (t *Tagger) updateVolumes() error {
-	if t.ebsVolume == nil {
-		t.ebsVolume = NewEbsVolume()
+/*
+	Retrieve metadata from IMDSv2,IMDSv1 and use these metadata to:
+	* Extract InstanceID, ImageID, InstanceType to create custom dimension for collected metrics
+	* Extract InstanceID to retrieve Instance's Volume and Tags
+	* Extract Region to create aws session with custom configuration
+	For more information on IMDSv2, please follow this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html
+*/
+func (t *Tagger) initialRetrievalOfEC2Metadata() error {
```

Comment:
This function name is confusing. What is the purpose of this function? It reaches out to IMDS and then stores some of the metadata so that we can use it internally. There is never really any reason to have `Of` in a function or method name. That is a clear indication that the naming is improper. The name should be changed to be more clear about the purpose. "Initial retrieval of EC2 Metadata" doesn't tell me what this does clearly. Why is it "initial"? 

```suggestion
func (t *Tagger) cacheEC2Metadata() error {
```

Reply:
Agree. Should go with the initial name that you recommended me `deriveEC2MetadataFromIMDS` instead of `initialRetrievalOfEC2Metadata`. Thanks for reminding me.